x-pack/plugins/cases/public/components/all_cases/columns_popover.tsx

```diff
@@ -81,38 +111,55 @@ export const ColumnsPopover: React.FC<Props> = ({
           {i18n.COLUMNS}
         </EuiButtonEmpty>
       }
-      isOpen={isPopoverOpen}
-      closePopover={closePopover}
-      panelPaddingSize="none"
-      anchorPosition="leftUp"
-      hasDragDrop
-      zIndex={0}
-      data-test-subj="column-selection-popover"
     >
+      <EuiPopoverTitle>
+        <EuiFieldText
+          compressed
+          placeholder={i18n.SEARCH}
+          aria-label={i18n.SEARCH_COLUMNS}
+          value={columnSearchText}
+          onChange={(e: ChangeEvent<HTMLInputElement>) =>
+            setColumnSearchText(e.currentTarget.value)
+          }
+          data-test-subj="column-selection-popover-search"
+        />
+      </EuiPopoverTitle>
       <EuiDragDropContext onDragEnd={onDragEnd}>
-        <EuiFlexGroup css={{ width: 300 }}>
+        <EuiFlexGroup
+          css={{ width: 300 }}
+          data-test-subj="column-selection-popover-drag-drop-context"
+        >
           <EuiFlexItem>
             <EuiDroppable
               droppableId="casesColumnDroppableArea"
-              css={{ paddingBottom: euiTheme.size.base }}
+              css={
+                isDragEnabled
+                  ? {
+                      paddingBottom: euiTheme.size.base,
+                    }
+                  : {
+                      paddingTop: euiTheme.size.base,
+                    }
+              }
             >
-              {selectedColumns.map(({ field, name, isChecked }, idx) => (
+              {filteredColumns.map(({ field, name, isChecked }, idx) => (
                 <EuiDraggable
                   key={field}
                   index={idx}
                   draggableId={field}
-                  customDragHandle={true}
-                  hasInteractiveChildren={true}
+                  isDragDisabled={!isDragEnabled}
                   css={{ height: euiTheme.size.xl, paddingLeft: euiTheme.size.base }}
+                  customDragHandle
+                  hasInteractiveChildren
                 >
                   {(provided) => (
                     <EuiFlexGroup alignItems="center" gutterSize="m" justifyContent="spaceBetween">
-                      <EuiFlexItem>
+                      <EuiFlexItem grow={false}>
                         <EuiSwitch
                           label={name}
```

Comment:
> Finally, the EuiSwitch label does not allow us to highlight the search text like [in here](https://eui.elastic.co/#/forms/selectable#searchable) for example so I did not include it. If we decide it is worth the effort I can spend some time playing with the components to find an alternative solution.
 
You can use `label={<EuiHighlight search={columnSearchText}>{name}</EuiHighlight>}`

Reply:
Ah, I didn't realize the `label` prop allowed for a component, thanks!

Reply:
This worked 👌 Additionally I tried to include the [.eui-textTruncate](https://eui.elastic.co/#/theming/typography/utilities#wrapping) to get rid of the styles in `labelProps` but that didn't work 😞